admin/src/components/Wysiwyg/index.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ *
+ * WYSIWYG
+ *
+ */
+
+import React, { useRef, useState } from 'react';
+//@ts-ignore
+import styled, {ThemeInterface} from 'styled-components';
+import { useIntl } from 'react-intl';
+//@ts-ignore
+import { Typography } from '@strapi/design-system/Typography';
+//@ts-ignore
+import { Box } from '@strapi/design-system/Box';
+//@ts-ignore
+import { Stack } from '@strapi/design-system/Stack';
+//@ts-ignore
+import { prefixFileUrlWithBackendUrl, useLibrary } from '@strapi/helper-plugin';
+import Editor from './Editor';
+import WysiwygNav from './WysiwygNav';
+import WysiwygFooter from './WysiwygFooter';
+import Hint from '../Hint';
+import {
+  markdownHandler,
+  listHandler,
+  titleHandler,
+  insertFile,
+  quoteAndCodeHandler,
+} from './utils/utils';
+import { EditorLayout } from './EditorLayout';
+
+
+const LabelAction = styled(Box)`
+  svg path {
+    fill: ${({ theme }:ThemeInterface) => theme.colors.neutral500};
+  }
+`;
+
+const TypographyAsterisk = styled(Typography)`
+  line-height: 0;
+`;
+
+type WysiwygProps = {
+  description?:{
+    id: string,
+    defaultMessage: string,
+    values: {},
+  },
+  disabled?: boolean,
+  error?: string,
+  intlLabel:{
+    id: string,
+    defaultMessage: string,
+    values: {},
+  },
+  labelAction?: JSX.Element,
+  name: string,
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
+  placeholder?: {
+    id: string,
+    defaultMessage: string,
+    values: {},
+  },
+  required?: boolean,
+  value: string,
+};
+
+const Wysiwyg: React.FC<WysiwygProps> = ({
+  description,
+  disabled,
+  error,
+  intlLabel,
+  labelAction,
+  name,
+  onChange,
+  placeholder,
+  value,
+  required,
+}) => {
+  const { formatMessage } = useIntl();
+  const textareaRef = useRef<string | null>(null);
+  const editorRef = useRef<string | null>(null);
+  const [isPreviewMode, setIsPreviewMode] = useState<boolean>(false);
+  const [mediaLibVisible, setMediaLibVisible] = useState<boolean>(false);
+  const [isExpandMode, setIsExpandMode] = useState<boolean>(false);
+  const { components } = useLibrary();
+
+  const MediaLibraryDialog = components['media-library'];
+
+  const handleToggleMediaLib = () => setMediaLibVisible(prev => !prev);
+  const handleTogglePreviewMode = () => setIsPreviewMode(prev => !prev);
+  const handleToggleExpand = () => {
+    setIsPreviewMode(false);
+    setIsExpandMode(prev => !prev);
+  };
```

Comment:
```suggestion
  const handleToggleMediaLib = () => setMediaLibVisible(negate);
  const handleTogglePreviewMode = () => setIsPreviewMode(negate);
  const handleToggleExpand = () => {
    setIsPreviewMode(false);
    setIsExpandMode(negate);
  };
```

Reply:
And `useCallback`, please.

Reply:
wysiwyg should stays as is, in case of changes from Strapi side